### ISSUE TITLE
More features for `mapset`

### ISSUE THREAD

**Issue by raph-m** [zyedidia/generic#21]
Hello,

We have our own implementation of sets in our codebase (which is not relying on generics), and we would like to switch to a generics aproach for this.

The `mapset` package looks great but we can't use this as is without adding more features.
Here are some methods that we use regularly and that we would need.

```go
// returns items of the set in a slice
func (s Set[K]) Items() []K

// creates a new set from a slice
func FromSlice[K]([]K) Set[K]

func (s Set[K]) Copy() Set[K]

// set with all items that are in both s and other
func (s Set[K]) Intersection(other Set[K]) Set[K]

// set with all items that are in either s or other
func (s Set[K]) Union(other Set[K]) Set[K]

// all items that are in s but not in other
func (s Set[K]) Substraction(other Set[K]) Set[K]

// are the sets equal
func (s Set[K]) Equals(other Set[K]) bool

// and maybe this one
// returns true if all items of other are included in s
func (s Set[K]) IsSuperSet(other Set[K]) bool
```

If you think that we should integrate those methods, I can take care of creating some PRs for this.

Cheers

**Comment by shlomi-dr** [comment]
I'd also add set intersection to this list.

**Comment by shlomi-dr** [comment]
Looks like https://github.com/zyedidia/generic/pull/8 does most of whats being asked here, only pending some docstrings before merge. 

If no one has time to do so, let me know and I will try to add them myself for the chance that it gets merged soon. 

One thing we may want to add are these (or some) functions to the Set interface.

**Comment by zyedidia** [comment]
I've merged #8 since it seems useful and doesn't seem like the author will provide another update. I think that should satisfy this use-case.
